Iterable/iterable-web-sdk
# Retain Authorization Type After updateUserEmail Invocations

## JIRA Ticket(s) if any

* [MOB-4388](https://iterable.atlassian.net/browse/MOB-4388)

## Description

Retains same authorization type even after `updateUserEmail` is called.

Previously, this was the functionality:

1. User authorizes with `setUserID`
2. User ID is attached to all outgoing requests
3. User calls `updateUserEmail` (which hits our backend to update the email on the user profile)
4. All new requests are authorized with the new email address and added to outgoing payloads, and newly generated JWT is signed with the new email address.

Now, authorization will work like:

1. User authorizes with `setUserID`
2. User ID is attached to all outgoing requests
3. User calls `updateUserEmail` (which hits our backend to update the email on the user profile)
4. All new requests continue to be authorized with the original user ID and added to outgoing payloads, and newly generated JWT is signed with the user ID.

## Test Steps

1. Run a local JWT generation backend by [cloning this repo](https://github.com/Iterable/jwt-generator) and running `docker-compose up -d`
6. Run example app in this repo with `yarn install:all && yarn start:all:react`
7. Open up `localhost:8080` in your browser and login with a user ID
   * any mock user ID will work fine
8. Navigate to /users page and fill out `updateUserEmail` form to change your email address 
9. Ensure in the network tab, the `/generate` endpoint has been called one last time with `{ userID: 'your-id' }` instead of email
10. Change the code in `react-example/src/index.tsx` to call `setEmail` on login again
11. Repeat steps 3-5, but this time ensure the payload passed to `/generate` is `{ email: 'your-new-email-address' }`
